lib/watch.js

```diff
@@ -47,46 +48,93 @@ function follows (ssb, state) {
 function newAbouts (ssb, state) {
   if (state.quiting) return
 
-  pull(
-    ssb.backlinks.read({
-      live: true,
-      old: false,
-      query: [{ $filter: {
-        dest: { $prefix: '@' },
-        value: { content: { type: 'about' } }
-      } }]
-    }),
-    pull.filter(msg => !msg.sync),
-    pull.filter(msg => {
-      // TODO schema validation!
-      return ref.isFeedId(msg.value.content.about) &&
-        (typeof msg.value.content.name === 'string' || msg.value.content.image)
-    }),
-    pull.drain(msg => {
-      state.updateQueue.add(msg.value.content.about)
-    })
-  )
+  function isValidAbout(msg) {
+    // TODO schema validation!
+    return ref.isFeedId(msg.value.content.about) &&
+      (typeof msg.value.content.name === 'string' || msg.value.content.image)
+  }
+
+  if (ssb.db) {
+    const { and, type, live, toPullStream } = ssb.db.operators
+    pull(
+      ssb.db.query(
+        and(type('about')),
+        live(),
+        toPullStream(),
+        pull.filter(isValidAbout),
+        pull.drain(msg => {
+          state.updateQueue.add(msg.value.content.about)
+        })
+      )
+    )
+  } else {
+    pull(
+      ssb.backlinks.read({
+        live: true,
+        old: false,
+        query: [{ $filter: {
+          dest: { $prefix: '@' },
+          value: { content: { type: 'about' } }
+        } }]
+      }),
+      pull.filter(msg => !msg.sync),
+      pull.filter(isValidAbout),
+      pull.drain(msg => {
+        state.updateQueue.add(msg.value.content.about)
+      })
+    )
+  }
 }
 
 function recentAuthors (ssb, state) {
   if (state.quiting) return
   const howFarBack = 300
 
-  pull(
-    pullCat([
-      ssb.createLogStream({ reverse: true, limit: howFarBack }),
-      ssb.createLogStream({ live: true, old: false })
-    ]),
-    pull.map(msg => msg.value.author),
-    pull.drain(author => {
-      if (!state.suggestCache[author]) {
-        state.updateQueue.add(author)
-      }
+  function updateState(author) {
+    if (!state.suggestCache[author]) {
+      state.updateQueue.add(author)
+    }
 
-      // update recent authors
-      var index = state.recentAuthors.indexOf(author)
-      if (~index) state.recentAuthors.splice(index, 1) // pop from current position
-      state.recentAuthors.push(author) // add to end
-    })
-  )
+    // update recent authors
+    var index = state.recentAuthors.indexOf(author)
+    if (~index) state.recentAuthors.splice(index, 1) // pop from current position
+    state.recentAuthors.push(author) // add to end
+  }
+
+  if (ssb.db) {
+    const { and, type, live, paginate, descending, toPullStream } = ssb.db.operators
+
+    pull(
+      pullCat([
+        pull(
+          ssb.db.query(
+            live({ old: true }),
+            paginate(howFarBack),
+            descending(),
+            toPullStream()
+          ),
+          pull.take(1)
+        ),
```

Comment:
I tested that it worked in both branches. Meaning we get the first 300 then all live updates. I think it would be easier to read without the live in the first one. Pushed up a commit for that.

Reply:
Yes it did seem to work just by looking at the code. But given that the semantics were slightly off, maybe a corner case bug could have slipped by. Thanks for changing it!